ocaml/testsuite/tools/expect_test.ml

```diff
@@ -313,8 +313,20 @@ let process_expect_file fname =
   let correction = eval_expect_file fname ~file_contents in
   write_corrected ~file:corrected_fname ~file_contents correction
 
-let repo_root = ref None
 let keep_original_error_size = ref false
+let preload_objects = ref []
+let main_file = ref None
+
+let read_anonymous_arg fname =
+  if Filename.check_suffix fname ".cmo"
+          || Filename.check_suffix fname ".cma"
+  then preload_objects := fname :: !preload_objects
+else
+  match !main_file with
+  | None -> main_file := Some fname
+  | Some _ ->
```

Comment:
It looks like the expect test runner used to be able to handle multiple files. (It ran `main` on each file.) Now it can only handle one. I'd prefer to do the same thing as upstream here. Maybe we can make the `cmo` and `cma` arguments be specified with a different flag, like `-load`, so that we can handle them differently.

Reply:
Did it? `main` calls `exit 0` after processing a file. Seems like before my change the runner just silently ignored the remaining arguments.

Reply:
Ah, OK, I think you're right. It may still be a simplification to require the cma/cmo arguments to be specified using a different flag instead of doing string munging, but I no longer feel as strongly.

Reply:
I copied this particular behavior from the OCaml toplevel, it also just checks the file extension.

Reply:
OK